sources/executor/execute_commands/execute_piped_command.c

```diff
@@ -22,9 +22,13 @@ t_validation	execute_piped_command(t_shell *shell, t_cmd *cmd)
 	cmd_path = cmd->cmd_path;
 	formatted_cmd = cmd->formatted_cmd;
 
+	if (is_builtin(shell->builtin_main, cmd, B_NUM_MAIN))
```

Comment:
Are these commands not meant to execute in the main thread?

Reply:
Correct, I have mentioned that in the comment -> bulletpoint 6